tests/aziot/aziot_test.go

```diff
@@ -0,0 +1,260 @@
+		"aziot-latest": "https://cloud-images.ubuntu.com/releases/22.04/release/ubuntu-22.04-server-cloudimg-amd64.img",
+	}
+	testScript = map[string]string{
+		"aziot-1.4.0":  "scripts/test_ubuntu20.04_aziot_1.4.0.sh",
+		"aziot-latest": "scripts/test_ubuntu22.04_aziot_latest.sh",
+	}
+	eveNode *tk.EveNode
+	// We need a shared access policy with the following permissions:
+	// Registration Status Read, Registration Status Write, Enrollment Read, Enrollment Write
+	// We can create a new policy in the Azure portal by going to :
+	// IoT Hub -> Device Provisioning Service (DPS) -> Shared access policies -> Add
+	// and then copy the connection string.
+	connectionString = os.Getenv("AZIOT_CONNECTION_STRING")
+	// The ID Scope is required to configure azure-iot in the VM,
+	// we can get it from the Azure IoT Hub -> Device Provisioning Service -> Overview
+	// and copy the "ID Scope".
+	aziotIdScope = os.Getenv("AZIOT_ID_SCOPE")
+)
+
+func TestMain(m *testing.M) {
+	log.Println("Azure IOT Hub Test started")
+	defer log.Println("Azure IOT Hub Test finished")
+
+	// check for secrets, if not available don't bother running the tests.
+	if connectionString == "" {
+		log.Fatalf("AZIOT_CONNECTION_STRING environment variable is not set")
+	}
+	if aziotIdScope == "" {
+		log.Fatalf("AZIOT_ID_SCOPE environment variable is not set")
+	}
+
+	node, err := tk.InitilizeTest(projectName, tk.WithControllerVerbosity("debug"))
+	if err != nil {
+		log.Fatalf("Failed to initialize test: %v", err)
+	}
+
+	eveNode = node
+	res := m.Run()
+	os.Exit(res)
+}
+
+func TestAzureIotTPMEndrolmentWithEveTools(t *testing.T) {
+	t.Log("TestAzureIotTPMEndrolmentWithEveTools started")
+	t.Log("Setup :\n\tAziot version 1.4.0 on Ubuntu-20.04-amd64\n\twith EVE-Tools and Proxy TPM")
+	defer t.Log("TestAzureIotTPMEndrolmentWithEveTools finished")
+
+	if !eveNode.EveIsTpmEnabled() {
+		t.Skip("TPM is not enabled, skipping test")
+	}
+
+	testAzureIotEdge(t, "aziot-1.4.0", false)
+}
+
+func TestAzureIotTPMEndrolmentWithVTPM(t *testing.T) {
+	t.Log("TestAzureIotTPMEndrolmentWithVTPM started")
+	t.Log("Setup :\n\tAziot (latest) on Ubuntu-22.04-amd64\n\twith VTPM")
+	defer t.Log("TestAzureIotTPMEndrolmentWithVTPM finished")
+
+	t.Skip("Skip test for now, it is failing!")
+
+	testAzureIotEdge(t, "aziot-latest", true)
+}
+
+func testAzureIotEdge(t *testing.T, version string, useVTPM bool) {
+	appName := tk.GetRandomAppName(projectName)
+	pubPorts := []string{sshPort + ":22"}
+	pc := tk.GetDefaultVMConfig(appName, tk.AppDefaultCloudConfig, pubPorts)
+	err := eveNode.EveDeployApp(appLink[version], pc,
+		tk.WithSSH(tk.AppDefaultSSHUser, tk.AppDefaultSSHPass, sshPort))
+	if err != nil {
+		t.Fatalf("Failed to deploy app: %v", err)
+	}
+	defer func() {
+		err = eveNode.AppStopAndRemove(appName)
+		if err != nil {
+			log.Errorf("Failed to stop and remove app: %v", err)
+		}
+	}()
+
+	// wait for the deployed app to appear in the list
+	time.Sleep(30 * time.Second)
```

Comment:
wasn't there a way to read app status directly from controller?

Reply:
yes, but in the first few seconds it return "app not found", even though it is deployed.

Reply:
see a few lines down : 

```
err = eveNode.AppWaitForRunningState(appName, 60*5)
	if err != nil {
		t.Fatalf("Failed to wait for app to start: %v", err)
	}
```

Reply:
huh, so that's a bug and a behaviour we don't want in Eden, I'll create a ticket for it then. Because it shouldn't be like that, right? :D

Reply:
I think so, let me check again and make sure it is happening consistently  or no.